earthpy/tests/test_spatial.py

```diff
@@ -37,3 +40,100 @@ def test_extent_to_json():
 
     with pytest.raises(AssertionError):
         es.extent_to_json([0, 1, 1, 0])
+
+
+# Create points GDF
+pts = np.array([[2, 2],
+                [3, 4],
+                [9, 8],
+                [-12, -15]])
+
+locs = [Point(xy) for xy in pts]
+locs_gdf = gpd.GeoDataFrame(locs,
+                            columns=['geometry'],
+                            crs={'init': 'epsg:4326'})
+
+# TODO? Create a create gdf from list function since we do it like 3
+# times in this test file already
+
+
+############## LINE OBJECTS #######
+
+linea = LineString([(1, 1), (2, 2), (3, 2), (5, 3)])
+lineb = LineString([(3, 4), (5, 7), (12, 2), (10, 5), (17, 17)])
+
+linez_gdf = gpd.GeoDataFrame([1, 2],
+                             geometry=[linea, lineb],
+                             crs={'init': 'epsg:4326'})
+
+############## POLY OBJECTS ################
+
+# Create crop box
+poly_inters = Polygon([(0, 0), (0, 10), (10, 10), (10, 0), (0, 0)])
+poly_in_gdf = gpd.GeoDataFrame([1],
+                             geometry=[poly_inters],
+                             crs={'init': 'epsg:4326'})
+
+# Shift by 20
+poly_out_gdf = poly_in_gdf.copy()
+poly_out_gdf = poly_in_gdf.geometry.apply(lambda x: shapely.affinity.
+                                          translate(x, xoff=20, yoff=20))
+# Buffer points to create multi poly
+locs_buff = locs_gdf.copy()
+locs_buff["geometry"] = locs_gdf.buffer(4)
+
+# Create geom w "donut hole"
+donut_geom = gpd.overlay(locs_buff, poly_in_gdf,
+                         how='symmetric_difference')
+
+multi_poly = donut_geom.unary_union
+multi_gdf = gpd.GeoDataFrame(gpd.GeoSeries(multi_poly),
+                   crs={'init': 'epsg:4326'})
+multi_gdf = multi_gdf.rename(columns={0: 'geometry'}).\
+    set_geometry('geometry')
+
+# LINE OBJECTS
+
+def test_clip_funs():
+    # Test that a bounding box returns error if the extents don't overlap
```

Comment:
You can break up this function into multiple smaller functions that have good names, and then you don't need so much narrative in the comments, e.g., 

`test_nonoverlapping_extents_error()`, `test_geometries_error()`, `test_clip_point()`, `test_clip_poly()`, `test_clip_multipoly()`, `test_clip_donut()`, `test_clip_line()`

If you want to have comments for these smaller functions, the docstrings are probably a good place! Breaking this out into multiple functions will also allow you to see which ones pass and fail, rather than having the whole thing fail if any one assertion fails.

Reply:
ok great. this was another question i had for you. answered here. i'd prefer to break it out like that just wasn't sure if the goal was a single function!